parsers/US_CA.py

```diff
@@ -34,37 +34,51 @@ def fetch_production(country_code='US-CA', session=None):
-
-    renewableResources = pandas.read_table(url, sep='\t\t', skiprows=2, header=None, names=['Hour', 'GEOTHERMAL','BIOMASS', 'BIOGAS', 'SMALL HYDRO','WIND TOTAL', 'SOLAR PV', 'SOLAR THERMAL'], skipfooter=27, skipinitialspace=True, engine='python')
-    otherResources = pandas.read_table(url, sep='\t\t', skiprows=30, header=None, names=['Hour', 'RENEWABLES', 'NUCLEAR', 'THERMAL', 'IMPORTS', 'HYDRO'], skipinitialspace=True, engine='python')
-
-    dailyData = []
+    url = 'http://www.caiso.com/outlook/SP/fuelsource.csv'
+    csv = pandas.read_csv(url)
+    latest_index = len(csv)-1
+    h, m = map(int, csv['Time'][latest_index].split(':'))
+    date = arrow.utcnow().to('US/Pacific').replace(hour = h, minute = m, second = 0, microsecond = 0)
+    production_map = {
+        'Solar': 'solar',
+        'Wind': 'wind',
+        'Geothermal': 'geothermal',
+        'Biomass': 'biomass',
+        'Biogas': 'biomass',
+        'Small hydro': 'hydro',
+        'Coal': 'coal',
+        'Nuclear': 'nuclear',
+        'Natural gas': 'gas',
+        'Large hydro': 'hydro',
+        'Other': 'unknown'
+    }
+    storage_map = {
+        'Batteries': 'battery'
+    }
+    data = {
+        'countryCode': country_code,
+        'production': {},
+        'storage': {},
+        'source': 'caiso.com',
+        'datetime': date.datetime
+    }
+    for key, value in production_map.items():
+        prod = csv[key][latest_index]
+        # if another mean of production created a value, sum them up
+        if value in data['production']:
+            data['production'][value] += prod
+        else:
+            data['production'][value] = prod
 
-    for i in range(0, 24):
-        data = {
-            'countryCode': country_code,
-            'production': {},
-            'storage': {},
-            'source': 'caiso.com',
-        }
-        data['production'] = {}
-        data['production']['biomass'] = renewableResources['BIOMASS'][i]
-        data['production']['gas'] = renewableResources['BIOGAS'][i]
-        data['production']['hydro'] = renewableResources['SMALL HYDRO'][i] + otherResources['HYDRO'][i]
-        data['production']['nuclear'] = otherResources['NUCLEAR'][i]
-        data['production']['solar'] = renewableResources['SOLAR PV'][i] + renewableResources['SOLAR THERMAL'][i]
-        data['production']['wind'] = renewableResources['WIND TOTAL'][i]
-        data['production']['geothermal'] = renewableResources['GEOTHERMAL'][i]
-        data['production']['unknown'] = otherResources['THERMAL'][i] # this is not specified in the list
-        # set the date at the end of the hour
-        data['datetime'] = yesterday.shift(hours = i+1).datetime
-        dailyData.append(data)
+    for key, value in storage_map.items():
+        storage = csv[key][latest_index]
+        # if another mean of storage created a value, sum them up
+        if value in data['production']:
+            data['storage'][value] += storage
```

Comment:
Are you sure storage is positive when energy is charged, and negative when it is discharged?

Reply:
Not 100% sure. This is not explicitly expressed. Should we ask them?
> An increasing amount of energy storage is being connected to the ISO grid, including non-generator resources, such as batteries, that can generate and store electricity until dispatched.

Reply:
it seems like a positive value means discharge:
![image](https://user-images.githubusercontent.com/1655848/34046870-c90bea76-e1ae-11e7-87b9-a5dde97586e4.png)
![image](https://user-images.githubusercontent.com/1655848/34046876-cdd61f86-e1ae-11e7-95e9-81efff9f3ca6.png)

This is the opposite convention of ours, so we need to multiply by -1